src/ui/EditTask.svelte

```diff
@@ -139,7 +147,7 @@
     const _onSubmit = () => {
         const { globalFilter } = getSettings();
         let description = editableTask.description.trim();
-        if (!description.includes(globalFilter)) {
+        if (addGlobalFilterOnSave) {
```

Comment:
What prevents this from adding a ` ` prefix when the global filter is empty? I see that it has fixed that problem that occurred in an earlier revision of this PR, but cannot work out why it works correctly now...

Reply:
When the global filter is empty there is nothing that would set `addGlobalFilterOnSave`.
In `onMount` the task description is not modified by `getDescriptionWithoutGlobalFilter`, and `indexOf(globalFilter)` is 0 (bigger than -1).